validphys2/src/validphys/filters.py

```diff
@@ -511,7 +558,7 @@ def __init__(self, initial_data: dict, *, defaults: dict, theory_parameters: dic
         self.rule_string = self.rule
         self.defaults = defaults
         self.theory_params = theory_parameters
-        ns = {*self.numpy_functions, *self.defaults, *self.variables, "idat", "central_value"}
+        ns = {*self.numpy_functions, *self.defaults.to_dict().keys(), *self.variables, "idat", "central_value"}
```

Comment:
```suggestion
        ns = {*self.numpy_functions, *self.defaults.to_dict(), *self.variables, "idat", "central_value"}
```

Not that it makes a difference though.